### PR TITLE
Add manual testing for delete and find

### PR DESCRIPTION
**Pull request by lancehaha**
image

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -639,13 +639,28 @@ testers are expected to do more *exploratory* testing.
...
 1. _{ more test cases …​ }_
 
+### Finding a person
+
+1. Finding with parameters missing.
+   1. Test case: `find` <br>
+      Expected: An error message is shown stating that the command format is incorrect and showing the user the correct format.
+   2. Test case: `find t/` <br>
+      Expected: Listing out everyone is the contact list.  
+2. Finding a person with all possible parameters specified.
```
Comment (EdaYeo) [inline docs/DeveloperGuide.md:649]: Should we add a Prerequisite: There is someone in the contact list with the name John Doe, phone number of 98765432, lives in room #08-0805 and has a "friends" tag here? Or alternatively, add another step to the test case, which is add n/John Doe p/98765432 e/johnd@example.com r/08-0805 t/friends

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -639,13 +639,28 @@ testers are expected to do more *exploratory* testing.
...
+   1. Test case:  `find n/John Doe p/98765432 r/08-0805 t/friends` <br>
+      Expected: Listing out the contact with a name John Doe, with a phone number of 98765432, and lives in the room 08-0805, who has a tag of friends.
```
Comment (EdaYeo) [inline docs/DeveloperGuide.md:651]: Should this be "Displaying the contact with name John Doe, phone number of 98765432, living in room #08-0805 and has a "friends" tag" instead?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -639,13 +639,28 @@ testers are expected to do more *exploratory* testing.
...
+   2. Test case: `find n/John Doe r/abcd` <br>
+      Expected: An error message is shown informing the user about the correct data format for ROOM_NUMBER.
 ### Saving data
 
 1. Dealing with missing/corrupted data files
 
    1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_
 
-1. _{ more test cases …​ }_
+2. _{ more test cases …​ }_
```
Comment (EdaYeo) [inline docs/DeveloperGuide.md:663]: Should we remove the {more test cases}?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -639,13 +639,28 @@ testers are expected to do more *exploratory* testing.
 
 1. _{ more test cases …​ }_
```
Comment (EdaYeo) [inline docs/DeveloperGuide.md:640]: Should we remove the {more test cases} here as well?

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -629,7 +629,7 @@ testers are expected to do more *exploratory* testing.
    1. Prerequisites: List all persons using the `list` command. Multiple persons in the list.
 
    1. Test case: `delete 1`<br>
-      Expected: First contact is deleted from the list. Details of the deleted contact shown in the status message. Timestamp in the status bar is updated.
+      Expected: Delete popup shows up. After confirming on the popup, first contact is deleted from the list. Details of the deleted contact shown in the status message. Timestamp in the status bar is updated.
```
Comment (EdaYeo) [inline docs/DeveloperGuide.md:632]: Do we still have a timestamp in the status bar? Is this some old implementation from AB3? (I might be missing it cause I never noticed a timestamp). If we do not have the timestamp anymore should we delete the last part?

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -629,7 +629,7 @@ testers are expected to do more *exploratory* testing.
...
 
    1. Test case: `delete 0`<br>
       Expected: No person is deleted. Error details shown in the status message. Status bar remains the same.
```
Comment (EdaYeo) [inline docs/DeveloperGuide.md:635]: If we do not have a status bar anymore should we remove the status bar part?

**Review by EdaYeo — COMMENTED** [review]
LGTM, just some nits!

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -639,13 +639,28 @@ testers are expected to do more *exploratory* testing.
 
 1. _{ more test cases …​ }_
 
+### Finding a person
+
+1. Finding with parameters missing.
```
Comment (SeanFoongjt) [inline docs/DeveloperGuide.md:644]: Should this be compulsory parameters instead of just parameters

Comment (lancehaha) [reply docs/DeveloperGuide.md:644]: no parameters are compulsory for `find` everything is optional using the term `compulsory parameter` will be misguiding, but there has to be some params

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -306,18 +306,19 @@ results based on the specified criteria.
 
 #### Implementation
 
-* the delete command is based on the index of the contact
+* The `delete` command extends `Command` and implements `Undoable`.
+* the `delete` command is based on the index of the contact.
```
Comment (EdaYeo) [inline docs/DeveloperGuide.md:310]: Should it be The with capital T here?

**Review by EdaYeo — COMMENTED** [review]
Just one last tiny nit, otherwise LGTM!

**Review by EdaYeo — APPROVED** [review]
LGTM!

**Pull request merged**